tap_bing_ads/__init__.py

```diff
@@ -735,13 +748,16 @@ async def poll_report(client, account_id, report_name, start_date, end_date, req
 def log_retry_attempt(details):
     LOGGER.info('Retrieving report timed out, triggering retry #%d', details.get('tries'))
 
+# retry the request for 5 times when Timeout error occurs
 @backoff.on_exception(backoff.constant,
-                      requests.exceptions.ConnectionError,
+                      (requests.exceptions.Timeout ,requests.exceptions.ConnectionError),
                       max_tries=5,
                       on_backoff=log_retry_attempt)
 def stream_report(stream_name, report_name, url, report_time):
     with metrics.http_request_timer('download_report'):
-        response = SESSION.get(url, headers={'User-Agent': get_user_agent()})
+        # Set request timeout with config param `request_timeout`.
+        request_timeout = get_request_timeout()
+        response = SESSION.get(url, headers={'User-Agent': get_user_agent()}, timeout=request_timeout)
```

Comment:
@namrata270998 Don't we have to add a similar timeout functionality to such code as well - https://github.com/singer-io/tap-bing-ads/pull/90/files#diff-8817c5cfeaccca0762dc1f3a6f1ceffe8df10485fb562505ae0f1a185bf66ae4R591

Please see other similar code updated in this PR - https://github.com/singer-io/tap-bing-ads/pull/90/files

Reply:
Yes, that's correct but as bing-ads uses its own SDK for API calls, we have added timeout here
https://github.com/singer-io/tap-bing-ads/pull/93/files#diff-8817c5cfeaccca0762dc1f3a6f1ceffe8df10485fb562505ae0f1a185bf66ae4R152